# Penguin Data

## import packages
import numpy as np

impoer pandas as pd

In [78]:
import numpy as np
import pandas as pd

## 1. read DataSet
use pandas

pd.read_csv('addres/name.csv')

In [79]:
data=pd.read_csv('penguins_size.csv')

# 2. find out shape of data

In [80]:
data_shaope = data.shape
data_size = data.size
data_ndim = data.ndim

print(f'data_shaope = {data_shaope} \ndata_size = {data_size} \ndata_ndim = {data_ndim}')

data_shaope = (344, 7) 
data_size = 2408 
data_ndim = 2


# 3. dataset info

## Information Of data
in the first print Information of data

there are 344 row and 7 column in the data

there are null value

In [81]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 7 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            344 non-null    object 
 1   island             344 non-null    object 
 2   culmen_length_mm   342 non-null    float64
 3   culmen_depth_mm    342 non-null    float64
 4   flipper_length_mm  342 non-null    float64
 5   body_mass_g        342 non-null    float64
 6   sex                334 non-null    object 
dtypes: float64(4), object(3)
memory usage: 18.9+ KB


In [82]:
data.head(7)

species     island  culmen_length_mm  culmen_depth_mm  flipper_length_mm  \
0  Adelie  Torgersen              39.1             18.7              181.0   
1  Adelie  Torgersen              39.5             17.4              186.0   
2  Adelie  Torgersen              40.3             18.0              195.0   
3  Adelie  Torgersen               NaN              NaN                NaN   
4  Adelie  Torgersen              36.7             19.3              193.0   
5  Adelie  Torgersen              39.3             20.6              190.0   
6  Adelie  Torgersen              38.9             17.8              181.0   

   body_mass_g     sex  
0       3750.0    MALE  
1       3800.0  FEMALE  
2       3250.0  FEMALE  
3          NaN     NaN  
4       3450.0  FEMALE  
5       3650.0    MALE  
6       3625.0  FEMALE

In [83]:
data.tail()

species  island  culmen_length_mm  culmen_depth_mm  flipper_length_mm  \
339  Gentoo  Biscoe               NaN              NaN                NaN   
340  Gentoo  Biscoe              46.8             14.3              215.0   
341  Gentoo  Biscoe              50.4             15.7              222.0   
342  Gentoo  Biscoe              45.2             14.8              212.0   
343  Gentoo  Biscoe              49.9             16.1              213.0   

     body_mass_g     sex  
339          NaN     NaN  
340       4850.0  FEMALE  
341       5750.0    MALE  
342       5200.0  FEMALE  
343       5400.0    MALE

In [84]:
data["culmen_depth_mm"].describe

<bound method NDFrame.describe of 0      18.7
1      17.4
2      18.0
3       NaN
4      19.3
       ... 
339     NaN
340    14.3
341    15.7
342    14.8
343    16.1
Name: culmen_depth_mm, Length: 344, dtype: float64>

# 4. unique Species

there are null (nan) and Wrong data ( . ) at sex of penguins

In [85]:
# species of penguins
print(f'species of penguins is ----{data['species'].unique()}')
print(f'island of penguins is -----{data['island'].unique()}')
print(f'sex of penguins is --------{data['sex'].unique()}')

species of penguins is ----['Adelie' 'Chinstrap' 'Gentoo']
island of penguins is -----['Torgersen' 'Biscoe' 'Dream']
sex of penguins is --------['MALE' 'FEMALE' nan '.']


In [86]:
data[data.isnull().any(axis='columns')]


species     island  culmen_length_mm  culmen_depth_mm  flipper_length_mm  \
3    Adelie  Torgersen               NaN              NaN                NaN   
8    Adelie  Torgersen              34.1             18.1              193.0   
9    Adelie  Torgersen              42.0             20.2              190.0   
10   Adelie  Torgersen              37.8             17.1              186.0   
11   Adelie  Torgersen              37.8             17.3              180.0   
47   Adelie      Dream              37.5             18.9              179.0   
246  Gentoo     Biscoe              44.5             14.3              216.0   
286  Gentoo     Biscoe              46.2             14.4              214.0   
324  Gentoo     Biscoe              47.3             13.8              216.0   
339  Gentoo     Biscoe               NaN              NaN                NaN   

     body_mass_g  sex  
3            NaN  NaN  
8         3475.0  NaN  
9         4250.0  NaN  
10        3300.0  NaN  
11        3700.0  NaN  
47        2975.0  NaN  
246       4100.0  NaN  
286       4650.0  NaN  
324       4725.0  NaN  
339          NaN  NaN

# 6.	find culmen_depth_mm in each island

In [103]:
data_back_fill = data.bfill()
print(data_back_fill.iloc[3])
print(data.iloc[4])
print(data.iloc[3])

species                 Adelie
island               Torgersen
culmen_length_mm          36.7
culmen_depth_mm           19.3
flipper_length_mm        193.0
body_mass_g             3450.0
sex                     FEMALE
Name: 3, dtype: object
species                 Adelie
island               Torgersen
culmen_length_mm          36.7
culmen_depth_mm           19.3
flipper_length_mm        193.0
body_mass_g             3450.0
sex                     FEMALE
Name: 4, dtype: object
species                 Adelie
island               Torgersen
culmen_length_mm           NaN
culmen_depth_mm            NaN
flipper_length_mm          NaN
body_mass_g                NaN
sex                        NaN
Name: 3, dtype: object


In [119]:
min_culmen_length_mm = data.groupby('island')['culmen_length_mm'].min()
max_culmen_length_mm = data.groupby('island')['culmen_length_mm'].max()
median_culmen_length_mm = data.groupby('island')['culmen_length_mm'].median()
mean_culmen_length_mm = data.groupby('island')['culmen_length_mm'].mean()
print(f'-----min_culmen_length_mm-----\n {min_culmen_length_mm}')
print(f'-----max_culmen_length_mm-----\n {max_culmen_length_mm}')
print(f'-----median_culmen_length_mm-----\n {median_culmen_length_mm}')
print(f'-----mean_culmen_length_mm-----\n {mean_culmen_length_mm}')


-----min_culmen_length_mm-----
 island
Biscoe       34.5
Dream        32.1
Torgersen    33.5
Name: culmen_length_mm, dtype: float64
-----max_culmen_length_mm-----
 island
Biscoe       59.6
Dream        58.0
Torgersen    46.0
Name: culmen_length_mm, dtype: float64
-----median_culmen_length_mm-----
 island
Biscoe       45.80
Dream        44.65
Torgersen    38.90
Name: culmen_length_mm, dtype: float64
-----mean_culmen_length_mm-----
 island
Biscoe       45.257485
Dream        44.167742
Torgersen    38.950980
Name: culmen_length_mm, dtype: float64


# 7.	body mass in each sex

In [201]:
data_back_fill = data.bfill()
print(f'befor clean data {data_back_fill['sex'].unique()}')
# Change value  sex = "."
data_back_fill['sex'] = data_back_fill['sex'].replace(".", "MALE")
print(f'after clean data {data_back_fill['sex'].unique()}')

#----------------------------------------
body_mass_sex = data_back_fill.groupby('sex')['body_mass_g'].mean()
print(f'body mass in each sex {body_mass_sex}')

befor clean data ['MALE' 'FEMALE' '.']
after clean data ['MALE' 'FEMALE']
body mass in each sex sex
FEMALE    3856.250000
MALE      4546.656977
Name: body_mass_g, dtype: float64
